src/client/metric/countvector.rs

```diff
@@ -0,0 +1,149 @@
+use super::*;
+
+/// A count vector for multiple strictly increasing integer values, in possibly
+/// varying increments
+///
+/// Internally uses an `InstanceMetric<u64>` with `Semantics::Counter` and
+/// `Count::One` scale, and `1` count dimension
+pub struct CountVector {
+    im: InstanceMetric<u64>,
+    indom: Indom,
+    init_val: u64
+}
+
+impl CountVector {
+    /// Creates a new count vector with given initial value and instances
+    pub fn new(name: &str, init_val: u64, instances: &[&str],
```

Comment:
see that's the thing here, the point of a CountVector is to provide a simple means to get an instance metric that outputs values based on count semantics. We probably don't want to give `set_val` to the user on this type, but at the same time, the user might want to start counting one instance at `0` and another at `1`. Again InstanceMetric is a set of metrics with a common metadata, but since their values can be different, so can initial values. We can set them all to one initial value and then allow the user to do `m.set_val("instance2", 1)`, but that would mean giving `set_val` on the type, which only creates bloat, and if the user wanted that functionality, he would create an InstanceMetric instead of a CounterVector